### ISSUE TITLE
Directory/file completion broken for files/dirs with spaces

### ISSUE THREAD

**Issue by gevhaz** [iterative/shtab#74]
## The problem

If I use `shtab.DIRECTORY` or `shtab.FILE` and try to complete a directory with spaces in the name, I just get a jumble of words as suggestions and the directory name isn't completed.

## How to reproduce

Here is the relevant part of the python code

```python
        parser.add_argument(
            "-d",
            "--directory",
            action="store",
            required=True,
            dest="dir",
            help="directory",
        ).complete = shtab.DIRECTORY  # type: ignore
```

Create a directory with spaces in the name:

```bash
$ mkdir "dir with many spaces"
$ ls
dir with many spaces
```

Completion without typing part of the word first yields a list of the words in the directory name:

```bash
$ my_script -d
dir     many    spaces  with
```

Nothing is completed when pressing tab.

Add the start of the directory name:

```bash
$ my_script -d di
dir     many    spaces  with
```

Same result.

## More info

Compgen, seems to handle it just fine:

```bash
$ compgen -d
dir with many spaces
$ compgen -d dir
dir with many spaces
```

Here's a relevant part of the generated completion script:

```bash
_shtab_my_script__d_COMPGEN=_shtab_compgen_dirs
...
# $1=COMP_WORDS[1]
_shtab_compgen_dirs() {
  compgen -d -- $1  # recurse into subdirs
}
```

So I can't see where it goes wrong. Seems to work fine when I do it manually. I'm happy to test stuff if you have suggestions.

**Comment by casperdcl** [comment]
Yup, looks like a bug

**Comment by casperdcl** [comment]
Should be fixed in [`shtab>=1.5.5`](https://github.com/iterative/shtab/releases/tag/v1.5.5)

**Comment by gevhaz** [comment]
Thank you! It works :)
